File: golang/876-middle-of-the-linked-list.go
package golang

import "math"

type ListNode struct {
	Val  int
	Next *ListNode
}

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func middleNode(head *ListNode) *ListNode {
    current, len := head, 0
    for current != nil {
        len++
        current = current.Next
    }
    
    current = head
    i, targetIdx := 1, int(math.Floor(float64(len/2))) + 1
    for i < targetIdx {
        i++
        current = current.Next
    }
    return current
}

File: typescript/876-middle-of-the-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

 function middleNode(head: ListNode | null): ListNode | null {
    let current = head, len = 0
    while (current) {
        len++
        current = current.next
    }
    
    current = head
    let i = 1
    const targetIdx = Math.floor(len/2) + 1
    while (i++ < targetIdx) {
        current = current.next
    }
    return current
};